### DISCUSSION TITLE
After adding custom domain there are now 404 errors. Using Google and Project Pages

### DISCUSSION THREAD

**Discussion by jculilap** [community/community#21570] [Other Feature Feedback, Questions, & Ideas]
<p>Hello,</p>
<p>I’ve been looking at many different threads here and online but I can’t seem to get my problem fixed.</p>
<p>Some background:</p>
<p>My page was originally working as intended under <em><a href="http://username.github.io/projectname" rel="nofollow noopener">username.github.io/projectname</a></em> but after I added a custom domain it brings up a blank page and going into the console it says I have 404 errors. I also have the CNAME file included too.</p>
<p>Custom domain name is bought from Google and I am using a project page repo and not the profile one.</p>
<p><a href="https://github.com/shiniko/Portfolio/" rel="nofollow noopener">My repo.</a></p>
<p>(I’ve been making a lot of changes trying to fix this but I hope it’s not messing it up. I did wait 24 hours after first inputing the custom domain name but it seems to still give the same errors, thus all the changes you see now is currently taking place.)</p>
<p>My google domain is set up like so. </p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/bd3dc8eee802df6b8a1fbbd19aab3226d9938f88.png" data-download-href="/uploads/short-url/r06C9LfKfDSKcXJl4a8qNC096Os.png?dl=1" title="image.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/16933392/181065648-2b0a0b06-590f-4d10-806f-bb049c24fe5e.png" alt="image.png" data-base62-sha1="r06C9LfKfDSKcXJl4a8qNC096Os" width="690" height="272" srcset="https://aws1.discourse-cdn.com/github/optimized/1X/bd3dc8eee802df6b8a1fbbd19aab3226d9938f88_2_690x272.png, https://aws1.discourse-cdn.com/github/optimized/1X/bd3dc8eee802df6b8a1fbbd19aab3226d9938f88_2_1035x408.png 1.5x, https://aws1.discourse-cdn.com/github/original/1X/bd3dc8eee802df6b8a1fbbd19aab3226d9938f88.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/bd3dc8eee802df6b8a1fbbd19aab3226d9938f88_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image.png</span><span class="informations">1064×420 14.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>and my github settings:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/b0fa46d3815e9df448dc9b8bd39588e5c320e881.png" data-download-href="/uploads/short-url/pfCfvja36ZW9jPbQ3T0dhr1HS6t.png?dl=1" title="image.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/16933392/181065649-4c96f2b4-d852-4b34-b509-5b59714923af.png" alt="image.png" data-base62-sha1="pfCfvja36ZW9jPbQ3T0dhr1HS6t" width="690" height="480" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/b0fa46d3815e9df448dc9b8bd39588e5c320e881_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image.png</span><span class="informations">701×488 20.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>When I go to the website it opens up a blank page, I open up the console on “Network” and shows this:  </p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/c770b13c654e3be1b52c420fa2c97a544d46535e.png" data-download-href="/uploads/short-url/ssksrHdH3BThC9uPPzT96ckE1MW.png?dl=1" title="image.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/16933392/181065650-a262abc8-9374-42f8-ae84-10bad18bab0d.png" alt="image.png" data-base62-sha1="ssksrHdH3BThC9uPPzT96ckE1MW" width="690" height="173" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/c770b13c654e3be1b52c420fa2c97a544d46535e_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image.png</span><span class="informations">990×249 11.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>So I’m assuming that since it’s reading the html file then Google must be working correctly and there’s something going on with my repo’s references, right? </p>
<p>When I click on each error I can see the request url is not including the docs/ folder in the path file. So I tried manually adding in the path but that still doesn’t work.</p>
<p>I’ve been trying to edit the docs/404.html and docs/index.html pages by changing either the href or src but no luck.</p>
<p>Most of the other tutorials online seem to only refer to the github profile version of _username.github.io _but some of them mentioned that it should work for both profile and project pages on the google domain.</p>
<p>I’ve been stuck on this for hours now :frowning:</p>

**Comment by justfiji** [comment]
<p>In your package.json file make sure the “homepage” variable is set to the same custom domain as well</p>

**Comment by lee-dohm** [comment]
<p>A CNAME record is required if you want to use <a href="https://help.github.com/en/articles/about-supported-custom-domains#subdomains" rel="nofollow noopener">a subdomain</a> for hosting your GitHub Pages site, for example <code>[www.example.com](http://www.example.com)</code>. The <code>www</code> part is the subdomain. It appears that you want to use <a href="https://help.github.com/en/articles/about-supported-custom-domains#apex-domains" rel="nofollow noopener">an apex domain</a>, for example <code>example.com</code>. For an apex domain, a CNAME can’t be used and you have to <a href="https://help.github.com/en/articles/setting-up-an-apex-domain" rel="nofollow noopener">use an ALIAS, ANAME, or A record</a> with your DNS provider.</p>
<p>From what it looks like in the screenshots you posted you have the following:</p>
<ul>
<li>Your DNS is set up to point <a href="http://www.jamielynnculilap.com" rel="nofollow noopener">www.jamielynnculilap.com</a> to <a href="http://shiniko.github.io" rel="nofollow noopener">shiniko.github.io</a>
</li>
<li>Your GitHub configuration is set up as if <a href="http://jamielynnculilap.com" rel="nofollow noopener">jamielynnculilap.com</a> is your host name,  <strong>not </strong> <a href="http://www.jamielynnculilap.com" rel="nofollow noopener">www.jamielynnculilap.com</a>
</li>
<li>If I request <a href="http://shiniko.github.io" rel="nofollow noopener">http://shiniko.github.io</a>, it gives the standard “There’s not a GitHub Pages site here” message (which is expected, given your configuration)</li>
<li>If I request <a href="http://shiniko.github.io/Portfolio" rel="nofollow noopener">http://shiniko.github.io/Portfolio</a>, I get redirected to <a href="https://jamielynnculilap.com" rel="nofollow noopener">https://jamielynnculilap.com</a> which gives the blank page you describe (which is also expected, given your configuration)</li>
<li>If I request <a href="https://jamielynnculilap.com" rel="nofollow noopener">https://jamielynnculilap.com</a>, I get the blank page you describe</li>
</ul>
<p>The contents of the blank page are:</p>
<pre><code>&lt;!doctype html&gt;
&lt;html lang="en"&gt;
&lt;head&gt;
  &lt;meta charset="utf-8"&gt;
  &lt;title&gt;PersWeb&lt;/title&gt;
  &lt;base href="/Portfolio/"&gt;
  &lt;meta name="viewport" content="width=device-width, initial-scale=1"&gt;
  &lt;link rel="icon" type="image/x-icon" href="favicon.ico"&gt;
&lt;link rel="stylesheet" href="styles.3647b3a2b47542ccb3ef.css"&gt;&lt;/head&gt;
&lt;body&gt;
  &lt;app-root&gt;&lt;/app-root&gt;
&lt;script type="text/javascript" src="runtime.26209474bfa8dc87a77c.js"&gt;&lt;/script&gt;&lt;script type="text/javascript" src="es2015-polyfills.66d0882ecaa572e181ea.js" nomodule&gt;&lt;/script&gt;&lt;script type="text/javascript" src="polyfills.8bbb231b43165d65d357.js"&gt;&lt;/script&gt;&lt;script type="text/javascript" src="main.9a1be62898eb19746491.js"&gt;&lt;/script&gt;&lt;/body&gt;
&lt;/html&gt;
</code></pre>
<p>And <a href="https://www.learnhowtoprogram.com/javascript/angular/angular-2-setup-root-component-root-module-and-more" rel="nofollow noopener"><code>\&lt;app-root\&gt;</code> is an Angular thing</a>, which it appears <a href="https://github.com/shiniko/Portfolio/blob/e5671030662d280e484b30fce29c475f74da2612/package.json#L14-L21" rel="nofollow noopener">your site is configured to use</a>.</p>
<p>Unfortunately, I’m not going to be much help with getting the Angular part to work. But it appears that the GitHub Pages stuff is all working as it should be.</p>
<p>I hope that helps! Let us know if you have more questions.</p>

**Comment by jculilap** [comment]
<p>Hi, I’m not quite sure what you mean by “homepage” variable, is this something I need to include myself in the package.json file? My package.jscon looks like <a href="https://github.com/shiniko/Portfolio/blob/master/package.json" rel="nofollow noopener">this</a>.</p>
<p>From what I found online it seems like the “homepage” variable refers to those using React, but I am currently trying to build my webpage with Angular.</p>

**Comment by jculilap** [comment]
<p>Hi, thank you for checking if my configuration is set up correctly.</p>
<p>So, correct me if I’m wrong, but I’m assuming since I have both the CNAME and A records configured it is somehow messing up how the webpage is showing right? So I removed the CNAME from my Google Domains (since I’m not using “www” for the subdomain) and kept the A record instead. </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/4/4909b62a14bfe9fa8bb91404ffc43c93a66a2a51.png" data-download-href="/uploads/short-url/aq7F3U5QJ3ajb4anZaDzzJI4ZcR.png?dl=1" title="image.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/16933392/181065655-bd2e02dd-015d-42ea-b2d6-9228732c49a3.png" alt="image.png" data-base62-sha1="aq7F3U5QJ3ajb4anZaDzzJI4ZcR" width="690" height="274" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/4/4909b62a14bfe9fa8bb91404ffc43c93a66a2a51_2_690x274.png, https://aws1.discourse-cdn.com/github/optimized/2X/4/4909b62a14bfe9fa8bb91404ffc43c93a66a2a51_2_1035x411.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/4/4909b62a14bfe9fa8bb91404ffc43c93a66a2a51.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/4/4909b62a14bfe9fa8bb91404ffc43c93a66a2a51_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image.png</span><span class="informations">1087×433 19 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>It still seems to be showing the same errors (and a white blank page) but I’ll wait a day and keep you updated if something changes.</p>
<p>What I’m more confused about is that if I <strong>do not use</strong> a custom domain name, and if I leave the site as <strong><a href="http://shiniko.github.io/Portfolio" rel="nofollow noopener">http://shiniko.github.io/Portfolio</a>, the website does show up correctly</strong>. So I thought I was doing the Github Pages configuration wrong.</p>
<p>If there is no other problems with how I configure Github, I am still stuck on what could be the problem :frowning:</p>
<p>I’ve had problems with gh-pages and master branch being the sources so I had to use docs/ as the source for my website to actually show up (without the domain name). And if I’m only using docs/ as the source then I have to use the project pages as my repo.</p>
<p>Is there anything else I can try? </p>

**Reply by maseelshah22** [reply]
were you able to find a solution for this? Leaving it as a github page works but adding a domain name is causing issues for me

**Comment by lee-dohm** [comment]
<p>Try creating a file named <code>test.html</code> in the root of your pages folder with the contents:</p>
<pre><code>&lt;!DOCTYPE html&gt;
&lt;html&gt;
  &lt;head&gt;
    &lt;meta charset="utf-8"&gt;
    &lt;title&gt;Test&lt;/title&gt;
  &lt;/head&gt;
  &lt;body&gt;
    &lt;p&gt;Test&lt;/p&gt;
  &lt;/body&gt;
&lt;/html&gt;
</code></pre>
<p>Wait for the pages build to go through, then navigate to <a href="https://jamielynnculilap.com/test.html" rel="nofollow noopener">https://jamielynnculilap.com/test.html</a>. This will test if it’s the custom domain configuration causing a problem or the Angular stuff causing a problem. If you don’t see the contents, then there’s something wrong with the domain configuration. If you do see the contents, then it’s the Angular stuff.</p>
<p>I hope that helps!</p>

**Comment by jculilap** [comment]
<p>hmm interesting, <a href="https://jamielynnculilap.com/test.html" rel="nofollow noopener">https://jamielynnculilap.com/test.html</a> does show up fine, so it is on Angular.</p>
<p>I’ll keep trying other solutions and update if I can get it to work. I am happy that the domain config is working as intended though!</p>

**Comment by jculilap** [comment, marked as answer]
<p>OKAY GOT IT! Oh my, that took way too long to figure out. I would like to thank <a class="mention" href="/lee-dohm">@lee-dohm</a> for his suggestion about doing the “test.html” and double checking that my domain configuration was setting up correctly, it helped me figure out what was the problem with the blank page showing.</p>
<p><strong>What I did to fix the blank page</strong> :</p>
<p>So for deploying to Github Pages I was following the Angular documentation shown below</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/4f7098e61ff7a32e7efc83d004a0f44e25341373.png" data-download-href="/uploads/short-url/bkKXrbB65UYF9a13hV7z2U7laTN.png?dl=1" title="image.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/16933392/181065660-b3f646da-6e36-4278-a9a3-b12a376f6852.png" alt="image.png" data-base62-sha1="bkKXrbB65UYF9a13hV7z2U7laTN" width="690" height="179" srcset="https://aws1.discourse-cdn.com/github/optimized/1X/4f7098e61ff7a32e7efc83d004a0f44e25341373_2_690x179.png, https://aws1.discourse-cdn.com/github/optimized/1X/4f7098e61ff7a32e7efc83d004a0f44e25341373_2_1035x268.png 1.5x, https://aws1.discourse-cdn.com/github/original/1X/4f7098e61ff7a32e7efc83d004a0f44e25341373.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/4f7098e61ff7a32e7efc83d004a0f44e25341373_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image.png</span><span class="informations">1251×326 27.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>This one and a bunch of other tutorials pretty much had the same concept, but after checking the “Network” status from the “test.html” file (that <a class="mention" href="/lee-dohm">@lee-dohm</a> mentioned) I was looking at how it was requesting the URL.</p>
<p><img src="https://user-images.githubusercontent.com/16933392/181065661-a4c2183d-8cb0-44c2-8893-fef9c3627671.png" alt="image.png" data-base62-sha1="vPtn4ugLElrAhJsTkYxOU6tQViV" width="668" height="245"></p>
<p>I realised that everytime the request was getting a 404 error the URL would look like: </p>
<p>https: <a href="//jamielynnculilap.com/" rel="nofollow noopener">//jamielynnculilap.com/</a>[name of repo]/[file name]</p>
<p>But the screenshot above doesn’t even include the repo name and it worked fine.</p>
<p>To fix this I built my project with</p>
<pre><code>ng build --prod --output-path docs
</code></pre>
<p>left base href as “/” which fixed the webpage and now it works! It’s quite strange because other people online seem to have the exact opposite problem (they needed to include the base href as their repository name), but my webpage seems to work as intended now. </p>
<p>Thanks for the help!</p>

**Reply by amydyson** [reply]
Thank you SO MUCH! I've been struggling with the same issue for A LONG time and FINALLY it now works.

**Comment by dreamyguy** [comment]
<p>This is what did it for me.</p>

**Comment by denisenanni** [comment]
<p>Did you set the output path to the folder ‘docs/’ in the file angular.json? Also when you build the project you need to indicate the output path.<br>
The index.html also will contain the URL of your project:<br>
&lt;title&gt;App title&lt;/title&gt;<br>
&lt;base href=“<a href="https://USERNAME.github.io/YOUR-REPO" rel="nofollow noopener">https://USERNAME.github.io/YOUR-REPO</a>” /&gt;<br>
Remember also to copy the index.html renaming it 404.html.<br>
Then you build the project with this command:<br>
ng build --prod --base-href <a href="https://USERNAME.github.io/REPOSITORY_NAME/" rel="nofollow noopener">https://USERNAME.github.io/REPOSITORY_NAME/</a> --output-path docs<br>
Then commit and push to origin the folder that has been created.</p>
<p>Sources:<br>
<a href="https://www.google.com/amp/s/codinglatte.com/amp/posts/angular/deploying-angular-apps-github-pages/" rel="nofollow noopener">https://www.google.com/amp/s/codinglatte.com/amp/posts/angular/deploying-angular-apps-github-pages/</a></p>

**Comment by EricPryzant** [comment]
<p>This is an old issue but I had the same problem and want to help anyone else that comes along later. I solved it using <code>ng deploy --base-href=https://&lt;URL&gt;.com --cname=&lt;URL&gt;.com</code> and changed the gh-pages location from docs/ to / (root). Solved all of my problems instantly.</p>

**Comment by dlozina** [comment]
<p>Thank you for this solution this really makes a difference!</p>

**Comment by moonsol124** [comment]
I had an exactly the same issue but in ReactJs. I also tested out with an 'test.html' file as mentioned in the comment sections adding it in the root of my git repository. I realized that my custom domain was working as intended. In my case the problem was the homepage address that i configured in my package.json file. The only thing that I had to do was changing the address to this format: 'http://{github ID}.github.io/' ...

**Reply by trumanjchan** [reply]
Thank you! I got my ReactJS site working with my domain with your comment. I followed the accepted answer [here](https://stackoverflow.com/questions/9082499/custom-domain-for-github-project-pages), and changed my package.json's homepage url to the format you had. After that I found out deploying that format change commit removed my CNAME in my gh-pages branch so I had to re-add the custom domain in the settings.

**Reply by ouattaraabraham** [reply]
with React changing the URL address of "homepage" to the same domain name example "homepage": "https//jamielynnculilap.com" solves the problem you also have to delete the browsing cache or change browser to test

**Comment by khuwender** [comment]
Hello everyone, 
I am facing an issue of how to track  user analytics of this amp page https://ezyschooling.com/amp/admissions/school-admission-in-bangalore-karnataka-boarding-schools in google analytics 4. to make the user experience more engaging.
